sdk-utils/src/main/java/com/redhat/parodos/sdkutils/RetryExecutorService.java

```diff
@@ -0,0 +1,87 @@
+package com.redhat.parodos.sdkutils;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+public class RetryExecutorService<T> implements AutoCloseable {
+
+	private final ExecutorService executor;
+
+	public RetryExecutorService() {
+		executor = Executors.newFixedThreadPool(1);
+	}
+
+	/**
+	 * Submit a task to the executor service, retrying on failure until the task succeeds
+	 * @param task The task to submit
+	 * @return The result of the task
+	 */
+	public T submitWithRetry(Callable<T> task) {
+		// @formatter:off
+		return submitWithRetry(task, () -> {}, () -> {}, 10 * 60 * 1000, 5000);
```

Comment:
@gciavarrini what should be the timeout? was it used to be 1 minute in previous implementation?

Reply:
In the bash implementation there were 100 retries.
I chose 1 minute because it seemed like a reasonable time interval and I didn't receive any feedback on this.
TBH, I'm not sure which is the best timeout value :)